app/uk/gov/hmrc/servicedependencies/service/TeamDependencyService.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.servicedependencies.service
+
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.servicedependencies.connector.TeamsAndRepositoriesConnector
+import uk.gov.hmrc.servicedependencies.controller.model.Dependencies
+import uk.gov.hmrc.servicedependencies.persistence.SlugInfoRepository
+import uk.gov.hmrc.servicedependencies.service.SlugDependenciesService.TargetVersion
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class TeamDependencyService @Inject()(
+  teamsAndReposConnector  : TeamsAndRepositoriesConnector,
+  slugInfoRepository      : SlugInfoRepository,
+  githubDepLookup         : DependencyDataUpdatingService,
+  serviceConfigsService   : ServiceConfigsService,
+  slugDependenciesService : SlugDependenciesService)(implicit ec: ExecutionContext)  {
+
+  def findAllDepsForTeam(team: String)(implicit hc: HeaderCarrier): Future[Seq[Dependencies]] =
+    for {
+       teamDetails     <- teamsAndReposConnector.getTeamDetails(team)
```

Comment:
Using the for comprehension with futures, these two calls (teamsAndReposConnector.getTeamDetails and githubDepLookup.getDependencyVersionsForAllRepositories) will execute in sequence rather than in parallel. To avoid this, this can be refactored to:

```
val futureTeamDetails = teamsAndReposConnector.getTeamDetails(team)
val futureGitHubDeps = githubDepLookup.getDependencyVersionsForAllRepositories

for {
  teamDetails <- futureTeamDetails
  githubDeps  <- futureGitHubDeps
  libs        =  teamDetails.libraries.map(l => githubDeps.find(_.repositoryName == l))
  ....
}
```

There's a good explanation of this here: https://stackoverflow.com/questions/19045936/scalas-for-comprehension-with-futures

Reply:
If the intention is to run in parallel, using `(teamDetails, githubDeps) <- (futureTeamDetails, futureGitHubDeps).mapN...` would make this explicit - rather than relying on Future implementation quirks which may get refactored.